builtins/amp-img.js

```diff
@@ -65,6 +65,16 @@ export class AmpImg extends BaseElement {
   }
 
   /** @override */
+  preconnectCallback(onLayout) {
+    // NOTE(@wassgha): since parseSrcset is computationally expensive and can
+    // not be inside the `buildCallback`, we went with preconnecting to the
+    // `src` url if it exists.
+    if (this.element.hasAttribute('src')) {
```

Comment:
When I'm stingy I'd write this as a single `getAttribute` for slightly less DOM access :)